src/lib/components/editor/LinkBibleReferenceButton.svelte

```diff
@@ -176,20 +178,30 @@
     async function addReference(startVerseId: string, endVerseId: string) {
         if (languageId === 1) {
             await postToApi('/resources/bible-references', {
-                resourceContentId,
+                resourceContentId: resourceContent.resourceContentId,
                 startVerseId: parseInt(startVerseId),
                 endVerseId: parseInt(endVerseId),
             });
         }
     }
 
+    function referenceExistsAlready(start: string, end: string) {
```

Comment:
What's the reasoning behind converting all of these ints to strings if we are comparing always comparing IDs as opposed to just comparing them as ints? Am I missing something?

Reply:
Yeah good question, the data in the DB is not super consistent so in some cases it'll be a number and in some cases a string. Normalizing to string before doing stuff has been the approach for safety. At some point we need to go through and make the data consistent probably.

Reply:
Ah, okay. Yeah, that seems like something we should address.